### PR TITLE
refactor CostTable.vue

### PR DESCRIPTION
**Pull request by zhouzilong2020**
make `CostTable.vue` independent to any data, now `CostTable.vue` would render any data passed to it. All the refresh logic is moved to `AppView.vue`.

This refactor makes `CostTable.vue` more reusable. I intend to add another table for only showing the item selected.

### CODE REVIEW

**Comment by zhouzilong2020** [comment]
PTAL

**Comment by d-bytebase** [comment]
there's conflict file frontend/src/components/CostTable.vue.

**Comment by zhouzilong2020** [comment]
PTAL

File: frontend/src/views/AppView.vue [hunk 1]
```diff
@@ -44,17 +45,18 @@
   <!-- dashboard -->
   <div class="mx-5 mt-5">
     <cost-table
-      :db-instance-list="dbInstanceStore.dbInstanceList"
-      :config="searchConfigStore.searchConfig"
+      :data-row="state.dataRow"
+      :isLoading="state.isLoading"
```
Comment (LiuJi-Jim) [inline frontend/src/views/AppView.vue:49]: use kebab-case (e.g. `:is-loading`) in property bindings too keep consistent

File: frontend/src/views/AppView.vue [hunk 2]
```diff
@@ -141,6 +153,148 @@ const copyURL = () => {
     });
 };
 
+const config = ref(searchConfigStore.searchConfig);
+watch(
+  () => config, // ref to searchConfigStore.searchConfig
```
Comment (LiuJi-Jim) [inline frontend/src/views/AppView.vue:158]: `watch(config, ...)`
we can watch config directly if it is already a Ref or computed obj.

File: frontend/src/components/costTable/CostTable.vue [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+]);
+
+const props = defineProps({
+  dataRow: {
+    type: Array as PropType<DataRow[]>,
+    default: () => [],
+  },
+  isLoading: { type: Boolean, default: () => false },
+  showEngineType: { type: Boolean, default: () => false },
+  showLeaseLength: { type: Boolean, default: () => false },
```
Comment (LiuJi-Jim) [inline frontend/src/components/costTable/CostTable.vue:253]: `default: false` is okay, when property's type is a primitive type (number, boolean, string)

**Review by LiuJi-Jim — COMMENTED** [review]
LG overall, with minor updates.

**Review by LiuJi-Jim — APPROVED** [review]
LGTM

**Pull request merged**